src/_33_4_Providing_the_function_to_solve.jl

```diff
@@ -1,19 +0,0 @@
-#!/usr/bin/env julia
```

Comment:
For underscore vs no underscore, I'm fine with whichever. I removed the underscore one since the no underscore ones looks more up-to-date.

For whether to keep the underscore one, I'm not sure what's the advantage of keeping incorrect or duplicated file around if they are not used.

For longer term, I personally don't think there's an advantage of having stricktly (i.e. with no manual editting) auto-generated wrappers. Reasons are:

* Unless the API is changing a lot between different versions,
  there isn't much advantage of keeping an auto-generated wrapper around.

    The main advantage of keeping an auto-generated wrapper is IMHO being able to generate it again, which is only necessary if you have to redo the API wrapping. If the API is not changing rapiddly, you shouldn't need to do this after the initial wrapper is generated.

    Hopefully, GSL doesn't break API as often as LLVM.....

* Unless the library is very (i.e. with no exceptions) consistent about their convension or such
  information is available in the metadata in some form (e.g. gobject-introspection),
  it is almost always necessary to deal with the memory management manually.
  Since a C pointer argument can be used for many purposes (in/out parameter, out parameter,
  array input, array output), it is very hard to generate the high level function automatically
  (and it's even harder for Clang.jl to do that since it doesn't even have the metadata you might
  find in the doc). I don't think it is a good idea to use the automatically generated wrapper as
  purely a wrapper of the `ccall` either since

    1. You have to wrap it with higher level code again

    2. You can very easily by-pass the protection mechanism in `ccall` and introduce memory
        corruption without even realizing it (Ref https://github.com/ihnorton/Clang.jl/issues/152)

Because of these, I think using Clang.jl (or for GSL.jl, the current wrappers generated from the doc) as a template and direction edit them into a higher level julia wrapper is a better way to go.